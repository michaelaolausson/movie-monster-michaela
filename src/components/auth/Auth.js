import React, { useEffect, useState } from 'react';
import { Container,Row, Col, Alert } from 'react-bootstrap'
import TMDB from '../../api/tmdb/TMDB'

const tmdb = new TMDB();
const BASE_URL_TMDB = "https://www.themoviedb.org/authenticate/"
const APP_HOST = "movie-monster-michaela.herokuapp.com"

const Auth = () => {

    const [ error, setError ] = useState(null)

    useEffect( () => {
        if (!localStorage.session_id ||localStorage.session_id === "undefined") {
          authRequest()
        } else {
          window.location.href = "/approved";
        }
    }, []);
    
    const authRequest = async () => {
        try {
            const response = await tmdb.getRequestToken();
            setError(null)
            if (window.location === "movie-monster-michaela.herokuapp.com" ) {
                window.location.href = `${BASE_URL_TMDB}${response.data.request_token}?redirect_to=https://${APP_HOST}/approved`;
            } else {
                window.location.href = `https://www.themoviedb.org/authenticate/${response.data.request_token}?redirect_to=http://5.57.244.204:3000/approved`;
            }    
        } catch {
            setError('Something went wrong logging you in. Sorry!')
        }
    }

    return (
        <Container>
            <Row>
                <Col>{error && <Alert id="error-box" variant="danger">{error}</Alert>}
                    <Alert variant="warning">You need to be authenticated. Stand by.</Alert>
                </Col>
            </Row>
        </Container>
 
    )
}

export default Auth;