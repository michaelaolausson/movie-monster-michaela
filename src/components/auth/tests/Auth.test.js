import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import Auth from "./../Auth";

describe("<Auth />", () => {
  const wrapper = shallow(<Auth />);
  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("should render error element on error", () => {
    wrapper.error = "Something is wrong";
    wrapper.find("#error-box");
  });
  it("should not render error element when state error equals null", () => {
    wrapper.error = null;
    wrapper.find("#error-box");
  });
});
