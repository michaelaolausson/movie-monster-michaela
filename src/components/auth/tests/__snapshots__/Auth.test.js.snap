// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should match the snapshot 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col\\"><div role=\\"alert\\" class=\\"fade alert alert-warning show\\">You need to be authenticated. Stand by.</div></div></div></div>"`;
